core_backend/app/languages/routers.py

```diff
@@ -0,0 +1,182 @@
+from typing import Annotated, List
+
+from fastapi import APIRouter, Depends
+from fastapi.exceptions import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ..auth.dependencies import get_current_fullaccess_user, get_current_readonly_user
+from ..auth.schemas import AuthenticatedUser
+from ..database import get_async_session
+from ..utils import setup_logger
+from .models import (
+    LanguageDB,
+    delete_language_from_db,
+    get_default_language_from_db,
+    get_language_from_db,
+    get_list_of_languages_from_db,
+    is_language_name_unique,
+    save_language_to_db,
+    update_language_in_db,
+)
+from .schemas import LanguageBase, LanguageRetrieve
+
+router = APIRouter(prefix="/languages")
+logger = setup_logger()
+
+
+@router.post("/", response_model=LanguageRetrieve)
+async def create_language(
+    language: LanguageBase,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> LanguageRetrieve | None:
+    """
+    Create language endpoint. Upsert language to PG database
+    """
+    language.language_name = language.language_name.upper()
+    if not (await is_language_name_unique(language.language_name, asession)):
+        raise HTTPException(status_code=400, detail="Language name already exists")
+
+    if language.is_default is True:
+        await unset_default_language(asession)
+
+    language_db = await save_language_to_db(language, asession)
+
+    return _convert_language_record_to_schema(language_db)
+
+
+@router.put("/{language_id}", response_model=LanguageRetrieve)
+async def edit_language(
+    language_id: int,
+    language: LanguageBase,
+    full_access_user: Annotated[
+        AuthenticatedUser, Depends(get_current_fullaccess_user)
+    ],
+    asession: AsyncSession = Depends(get_async_session),
+) -> LanguageRetrieve | None:
+    """
+    Edit language endpoint.
+    """
+    old_language = await get_language_from_db(
+        language_id,
+        asession,
+    )
+
+    if not old_language:
+        raise HTTPException(
+            status_code=404, detail=f"Language id `{language_id}` not found"
+        )
+    language.language_name = language.language_name.upper()
+    if (old_language.language_name != language.language_name) and (
+        not (await is_language_name_unique(language.language_name, asession))
+    ):
+        raise HTTPException(status_code=400, detail="Language name already exists")
+
+    if language.is_default:
+        await unset_default_language(asession)
+
+    updated_language = await update_language_in_db(
+        language_id,
+        language,
+        asession,
+    )
+
+    return _convert_language_record_to_schema(updated_language)
```

Comment:
Two comments here:

1. Should we split this up into two endpoints? One for changing the language name and one for setting as default. This might make more sense given the use-cases.
2. I think we should not give the option to set is_default to false because that means we are left with no language being default in the database. You should only be able to set a given language to default, since the app logic handles unsetting the previously default language.

Reply:
I think it is okay for now. But I am fine with doing more thinking and updates when we will actually build a frontend for languages management. Then we will have a clear understanding of what to do.